### ISSUE TITLE
prevent merging scripts when needed

### ISSUE THREAD

**Issue by hans2103** [joomla/joomla-cms#21907]
### Is your feature request related to a problem? Please describe.
Issue with adding structured data as json format to Joomla using PHP. 
When I add a script declaration of type = json it will be combined in the header. 

```
    $jsonscript                 = array();
    $jsonscript['website']      = '
{
    "@context": "http://schema.org/",
    "@type": "WebSite",
    "name": "ECMA"
    "url": "https://example.com/"
}';
    $jsonscript['organization'] = '
{
    "@context": "http://schema.org",
    "@type": "Organization",
    "url": "https://example.com/",
    "logo": "https://example.com/images/logo-example.png",
    "contactPoint": [
    {
        "@type": "ContactPoint",
        "telephone": "+31-6-11",
        "contactType": "customer service",
        "availableLanguage": ["Dutch", "English"]
    }
    ],
    "name": "ECMA"
}
';

    foreach ($jsonscript as $content) :
        $document->addScriptDeclaration($content, $type = 'application/ld+json');
    endforeach;
```

This will be rendered:
```
    <script type="application/ld+json">
{
    "@context": "http://schema.org/",
    "@type": "WebSite",
    "name": "ECMA",
    "url": "https://www.example.com/"
}
{
    "@context": "http://schema.org",
    "@type": "Organization",
    "url": "https://www.example.com",
    "logo": "https://example.com/images/logo-example.png",
    "contactPoint": [
    {
        "@type": "ContactPoint",
        "telephone": "+31-6-11",
        "contactType": "customer service",
        "availableLanguage": ["Dutch", "English"]
    }
    ],
    "name": "ECMA"
}
    </script>
```

Th rich snippets testing tool does not recognize the second json part. It will only recognize type = website.

### Describe the solution you'd like
It would be great if some kind of parameter can be added which will prevent merging the tags. 

The output I would like to see is:
```
    <script type="application/ld+json">
{
    "@context": "http://schema.org/",
    "@type": "WebSite",
    "name": "ECMA",
    "url": "https://www.example.com/"
}
    </script>
    <script type="application/ld+json">
{
    "@context": "http://schema.org",
    "@type": "Organization",
    "url": "https://www.example.com",
    "logo": "https://example.com/images/logo-example.png",
    "contactPoint": [
    {
        "@type": "ContactPoint",
        "telephone": "+31-6-11",
        "contactType": "customer service",
        "availableLanguage": ["Dutch", "English"]
    }
    ],
    "name": "ECMA"
}
    </script>
```

Comma separated entries seems not to work for the Rich Snippets Testing Tool

### Additional context
When this feature request is granted you are able to add json to the head of your website using the function addScriptDeclaration.

**Comment by ReLater** [comment]
Hm, I think you're doing something wrong by building your JSON strings manually instead of using json_encode($some_data_collection).

Why don't you collect your rich snippets datas in a (maybe multidimensional) $array first and in the end run a single json_encode($array); to get the correct JSON string.

`$document->addScriptDeclaration(json_encode($array), $type = 'application/ld+json');`

**Comment by ReLater** [comment]
Ah, sorry, now I understand your feature request.

I'm using `->addCustomTag(...)` in `foreach` instead of `->addScriptDeclaration(...)`

```
foreach (severalSnippetsCollections as $collected_data_array)
{
$output = array('<script type="application/ld+json">');
$output[] = json_encode($collected_data_array);
$output[] = '</script>';
JFactory::getDocument()->addCustomTag($output);
}
```

**Comment by hans2103** [comment]
@ReLater using the function `addCustomTag` can be used too, but the function `addScriptDeclaration` seems to be made for the job.

https://github.com/joomla/joomla-cms/blob/461eb0dc7016833c91f4a058f05aa147dc09586d/libraries/src/Document/Document.php#L571-L593

I'm trying to add a `script` with `$type="application/ld+json"` to the `head` of my Joomla site. 
The only missing part of this function is the ability to add the script separated from other scripts with the same type.

**Comment by SharkyKZ** [comment]
This is fixed in 4.0 with #25357. As the fix contains a B/C break, it won't make it to 3.x.

**Comment by ghost** [comment]
closed as stated above.
